dds/DCPS/transport/rtps_udp/MetaSubmessage.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_TRANSPORT_RTPS_UDP_METASUBMESSAGE_H
+#define OPENDDS_DCPS_TRANSPORT_RTPS_UDP_METASUBMESSAGE_H
+
+#include "Rtps_Udp_Export.h"
+
+#include "ConstSharedRepoIdSet.h"
+
+#include <dds/DCPS/RTPS/RtpsCoreTypeSupportImpl.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS
+{
+namespace DCPS
+{
+
+struct OpenDDS_Rtps_Udp_Export MetaSubmessage {
+  MetaSubmessage()
+    : from_guid_(GUID_UNKNOWN), dst_guid_(GUID_UNKNOWN), to_guids_(make_rch<ConstSharedRepoIdSet>()), redundant_(false) {}
+  MetaSubmessage(const RepoId& from, const RepoId& dst)
+    : from_guid_(from), dst_guid_(dst), to_guids_(make_rch<ConstSharedRepoIdSet>()), redundant_(false) {}
+  MetaSubmessage(const RepoId& from, const RepoId& dst, RcHandle<ConstSharedRepoIdSet> to)
```

Comment:
`dst` is "destination" and `to` is a list of GUIDs it will be sent to -- these are pretty similar concepts, so at least seeing this class in isolation may be confusing

Reply:
Right. The dst_guid_ field controls INFO_DST submessage, as well as determining which addresses to use, but to_guids_ is a list of guids whose addresses should also be used for the send, so for example for undirected heartbeats, we send to the addresses for guids of all associated readers. Also seems to get used for static discovery.

Are you suggesting a name change for to_guids_?

Reply:
I've renamed from_guid_ to src_guid_ and to_guids_ to addr_guids_ in an attempt to more clearly show what's going on.